### ISSUE TITLE
If statement in included files

### ISSUE THREAD

**Issue by mrozilla** [haoxins/gulp-file-include#111]
Would it be possible to include the if statement in the included files to check if the props are being passed down? Here's what I have in mind:

```
// parent.html
@@include('child.html', {
    "name": "John",
    "surname": "Doe"
}
```

```
// child.html
<div>@@name</div>
@@if (surname !== undefined) {
    <div>@@surname</div>
}
```

**Comment by zverevalexei** [comment]
Agreed. Ignore-undeclared-variables option would be awesome to have.
Pre-configuring each single variable makes a mess out of code and gives a slight chance for a bug to appear.

**Comment by sebastian-marinescu** [comment]
+1 

This is not possible in any way?

**Comment by kamlyuk** [comment]
Use **this**:
`<div>@@name</div>
@@if (**this.**surname) {
    <div>@@surname</div>
}`
